### ISSUE TITLE
Add GPG

### ISSUE THREAD

**Issue by d33tah** [anvaka/pm#9]
Here's a script you might find useful for that purpose:

https://gist.github.com/d33tah/9177395

**Comment by anvaka** [comment]
This sounds interesting! Just to clarify I understand everything right:

1. Each node in this graph is a person (user ID packet)
2. Each connection in this graph is signature (signature packet)

Is this accurate?

**Comment by d33tah** [comment]
Yup. You're going to download ~6GiB of data this way.

**Comment by anvaka** [comment]
Thanks! Do you know approximately how many nodes/edges in the final graph?

**Comment by d33tah** [comment]
I'd say hundreds of thousands, but I can't really remember.

**Comment by anvaka** [comment]
I see the script starts with `7909761da0a42745` and performs dfs on the graph. Does this guarantee that entire graph will be traversed? In other words - are there any isolated components in the graph?

**Comment by d33tah** [comment]
Yes, there are. This just describes the biggest island, sadly. You could get the isolated parts by getting mirrors of the key servers, but I never worked on that.

**Comment by anvaka** [comment]
I looked yesterday at their docs, and found this [[1]](https://pgp.mit.edu/emailhelp.html): 

> If you *REALLY* need the whole index file or key ring, *PLEASE* 
contact the administrator of a keyserver and ask them for a dump.

So I sent them email to bug-pks@mit.edu - haven't heard back yet.

**Comment by d33tah** [comment]
W dniu 08.09.2015 o 21:43, Andrei Kashcha pisze:
> I looked yesterday at their docs, and found this [[1]](https://pgp.mit.edu/emailhelp.html): 
> 
>> If you *REALLY* need the whole index file or key ring, *PLEASE* 
> contact the administrator of a keyserver and ask them for a dump.
> 
> So I sent them email to bug-pks@mit.edu - haven't heard back yet.

https://keyserver.mattrude.com/guides/building-server/#download-the-needed-database-files

That's quicker.
